File: app/src/main/java/com/ShaHar91/poems/ui/home/categories/CategoryComponent.java
package com.shahar91.poems.ui.home.categories;

import com.shahar91.poems.injection.ApplicationComponent;
import com.shahar91.poems.injection.scope.FragmentScope;
import com.shahar91.poems.ui.base.normal.BaseGoogleComponent;

import dagger.Component;

@FragmentScope
@Component(dependencies = ApplicationComponent.class)
public interface CategoryComponent extends BaseGoogleComponent<CategoryViewModel> {
    void inject(CategoryFragment categoryFragment);
}

File: app/src/main/java/com/ShaHar91/poems/ui/home/categories/CategoryComponent.kt
package com.shahar91.poems.ui.home.categories

import com.shahar91.poems.injection.ApplicationComponent
import com.shahar91.poems.injection.scope.FragmentScope
import com.shahar91.poems.ui.base.normal.BaseGoogleComponent
import dagger.Component

@FragmentScope
@Component(dependencies = [ApplicationComponent::class])
interface CategoryComponent : BaseGoogleComponent<CategoryViewModel?> {
    fun inject(categoryFragment: CategoryFragment?)
}